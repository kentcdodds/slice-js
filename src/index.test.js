
import path from 'path'
import spawn from 'spawn-command'
import glob from 'glob'

const BABEL_BIN_PATH = require.resolve('babel-cli/bin/babel-node')

const slicers = glob.sync(path.resolve(__dirname, '../test/fixtures/**/*.slicer.js'))

slicers.forEach(slicer => {
  test(relativeizePath(slicer), () => {
    return runBabelNode(slicer).then(stdout => {
      expect(stdout).toMatchSnapshot()
    })
  })
})

function runBabelNode(args = '', cwd = process.cwd()) {
  const isRelative = cwd[0] !== '/'
  if (isRelative) {
    cwd = path.resolve(__dirname, cwd)
  }

  return new Promise((resolve, reject) => {
    let stdout = ''
    let stderr = ''
    const command = `${BABEL_BIN_PATH} ${args}`
    const child = spawn(command, {cwd})

    child.on('error', error => {
      reject(error)
    })

    child.stdout.on('data', data => {
      stdout += data.toString()
    })

    child.stderr.on('data', data => {
      stderr += data.toString()
    })

    child.on('close', () => {
      if (stderr) {
        reject(stderr)
      } else {
        resolve(stdout)
      }
    })
  })
}

function relativeizePath(absolutePath) {
  return absolutePath.replace(process.cwd(), '.')
}
