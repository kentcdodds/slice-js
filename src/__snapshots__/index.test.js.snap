exports[`test ./test/fixtures/cond-expr.slicer.js 1`] = `
"./test/fixtures/cond-expr.js: consequent-only
    /* eslint import/prefer-default-export:0, no-nested-ternary:0, no-unneeded-ternary:0 */
    export { isConstantOrAlternate };

    function isConstantOrAlternate(a) {
      return \'isConsequent\';
    }
./test/fixtures/cond-expr.js: alternate-only
    /* eslint import/prefer-default-export:0, no-nested-ternary:0, no-unneeded-ternary:0 */
    export { isConstantOrAlternate };

    function isConstantOrAlternate(a) {
      return a === \'consequent\' ? \'isConsequent\' : \'isAlternate\';
    }
./test/fixtures/cond-expr.js: alternate\'s consequent
    /* eslint import/prefer-default-export:0, no-nested-ternary:0, no-unneeded-ternary:0 */
    export { isConstantOrAlternate };

    function isConstantOrAlternate(a) {
      return a === \'consequent\' ? \'isConsequent\' : a === \'alternate\' ? \'isAlternate\' : \`isAlternate\'s Consequent\`;
    }
./test/fixtures/cond-expr.js: both consequent and alternate
    /* eslint import/prefer-default-export:0, no-nested-ternary:0, no-unneeded-ternary:0 */
    export { isConstantOrAlternate };

    function isConstantOrAlternate(a) {
      return a === \'consequent\' ? \'isConsequent\' : a === \'alternate\' ? \'isAlternate\' : \`isAlternate\'s Consequent\`;
    }
./test/fixtures/cond-expr.js: consequent, alternate, and alternate\'s consequent
    /* eslint import/prefer-default-export:0, no-nested-ternary:0, no-unneeded-ternary:0 */
    export { isConstantOrAlternate };

    function isConstantOrAlternate(a) {
      return a === \'consequent\' ? \'isConsequent\' : a === \'alternate\' ? \'isAlternate\' : \`isAlternate\'s Consequent\`;
    }
"
`;

exports[`test ./test/fixtures/module.slicer.js 1`] = `
"./test/fixtures/module.js: multiply multiplies numbers together
    /*
     * this file may seem odd, but trust me, this is intended to help
     * ensure we\'re developing a good slice, not to have a good math
     * module :)
     */
    export { sum, multiply };

    function sum(a, b) {
      return a + b;
    }

    function multiply(a, b) {
      let product, i;
      product = 0;
      for (i = 0; i < b; i++) {
        product = sum(product, a);
      }
      return product;
    }
"
`;
